src/pipeline/src/etl/transform/transformer/greptime.rs

```diff
@@ -191,3 +197,214 @@ impl Transformer for GreptimeTransformer {
         &mut self.transforms
     }
 }
+
+fn resolve_schema(
+    index: Option<usize>,
+    value_data: ValueData,
+    column_schema: ColumnSchema,
+    row: &mut Vec<GreptimeValue>,
+    schema: &mut Vec<ColumnSchema>,
+) {
+    if let Some(index) = index {
+        let api_value = GreptimeValue {
+            value_data: Some(value_data),
+        };
+        let value_column_data_type = proto_value_type(&api_value);
+        // safety unwrap is fine here because index is always valid
+        let schema_column_data_type = schema.get(index).unwrap().datatype();
+        if value_column_data_type.is_some_and(|t| t != schema_column_data_type) {
+            row[index] = GreptimeValue { value_data: None };
```

Comment:
How about throwing an error here?

Reply:
I'm not sure which is better.

Reply:
I think raising an error is better when the column data type is changed.